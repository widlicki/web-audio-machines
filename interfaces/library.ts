import { Tier } from '.'

export interface AudioCategory {
  id: string
  name: string
  tier: Tier
}

export interface AudioFile {
  id: string
  name: string
  fileName: string
  category: string
  dateAdded: Date
  tags?: string[]
  tier: Tier
}
