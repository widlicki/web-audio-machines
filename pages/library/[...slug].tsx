import type { NextPage } from 'next'

const FilteredLibraryFile: NextPage = () => {
  return (
    <div>
      <h1>Filtered Library</h1>
    </div>
  )
}

export default FilteredLibraryFile
