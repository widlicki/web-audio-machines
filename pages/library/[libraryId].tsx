import type { NextPage } from 'next'

const LibraryFile: NextPage = () => {
  return (
    <div>
      <h1>Library File</h1>
    </div>
  )
}

export default LibraryFile
