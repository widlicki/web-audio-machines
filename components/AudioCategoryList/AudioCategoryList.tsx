import {
  Box,
  List,
  ListItem,
  ListItemButton,
  ListItemText,
} from '@mui/material'
import { useState } from 'react'
import { AudioCategory } from '../../interfaces/library'

export interface Props {
  categories: AudioCategory[]
}

const AudioCategoryList: React.FC<Props> = ({ categories }) => {
  const [selected, setSelected] = useState<string>('')

  const selectCat = (id: string) => {
    setSelected(id)
  }

  return (
    <Box>
      <List>
        {categories &&
          categories.map((cat) => (
            <ListItem key={cat.id}>
              <ListItemButton
                onClick={() => selectCat(cat.id)}
                selected={cat.id === selected}
              >
                <ListItemText primary={cat.name} />
              </ListItemButton>
            </ListItem>
          ))}
      </List>
    </Box>
  )
}

export default AudioCategoryList
