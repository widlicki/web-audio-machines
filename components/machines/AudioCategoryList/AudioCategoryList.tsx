import FolderIcon from '@mui/icons-material/Folder'
import FolderSpecialIcon from '@mui/icons-material/FolderSpecial'
import {
  Avatar,
  Box,
  List,
  ListItem,
  ListItemAvatar,
  ListItemButton,
  ListItemText,
} from '@mui/material'
import { useState } from 'react'
import { AudioCategory } from '../../../interfaces/library'
import { NotFoundMessage } from '../../../styles/Theme'

export interface AudioCategoryListProps {
  categories: AudioCategory[]
}

const AudioCategoryList = ({ categories }: AudioCategoryListProps) => {
  const [selected, setSelected] = useState<string>('')

  const selectCat = (id: string) => {
    setSelected(id)
  }

  return (
    <Box>
      <List>
        {categories && categories.length > 0 ? (
          categories.map((cat) => (
            <ListItem key={cat.id} sx={{ padding: 0 }}>
              <ListItemButton
                onClick={() => selectCat(cat.id)}
                selected={cat.id === selected}
              >
                <ListItemAvatar>
                  <Avatar>
                    {cat.tier === 'PREMIUM' ? (
                      <FolderSpecialIcon />
                    ) : (
                      <FolderIcon />
                    )}
                  </Avatar>
                </ListItemAvatar>
                <ListItemText primary={cat.name} />
              </ListItemButton>
            </ListItem>
          ))
        ) : (
          <NotFoundMessage>- No Data -</NotFoundMessage>
        )}
      </List>
    </Box>
  )
}

export default AudioCategoryList
