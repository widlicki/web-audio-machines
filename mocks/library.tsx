import { AudioCategory, AudioFile } from '../interfaces/library'

export const mockAudioCategories: AudioCategory[] = [
  {
    id: 'a0649ac4-9618-4dc0-993d-1dc4dcce1539',
    name: 'Kick',
    tier: 'FREE',
  },
  {
    id: 'feb9d1ce-5098-4a30-97bd-c248341615f4',
    name: 'Bass',
    tier: 'FREE',
  },
  {
    id: '998b77a6-25de-48c4-83ff-04e51cafeaaf',
    name: 'Synth',
    tier: 'PREMIUM',
  },
]

export const mockAudioFiles: AudioFile[] = [
  {
    id: '8f25cb68-7dd8-4335-9778-7950ea08a735',
    name: 'Kick-1',
    fileName: 'kick1.wav',
    category: 'Kick',
    dateAdded: new Date(2021, 12, 28),
    tags: ['kick', '808'],
    tier: 'FREE',
  },
  {
    id: 'c630edbe-59eb-4f90-ad08-b2448d3fa051',
    name: 'Kick-2',
    fileName: 'kick2.wav',
    category: 'Kick',
    dateAdded: new Date(2021, 12, 28),
    tags: ['kick', '909', 'house'],
    tier: 'FREE',
  },
  {
    id: '8a23c994-3c07-4276-ae48-531903d03005',
    name: 'Bass-1',
    fileName: 'bass1.wav',
    category: 'Bass',
    dateAdded: new Date(2021, 12, 28),
    tags: ['bass', 'minimal', 'techno'],
    tier: 'FREE',
  },
  {
    id: '7b506039-88c3-4f6a-8c64-75f89b9788f0',
    name: 'Synth-1',
    fileName: 'synth1.wav',
    category: 'Synth',
    dateAdded: new Date(2021, 12, 28),
    tags: ['synth', 'drone', 'cinematic'],
    tier: 'FREE',
  },
]
