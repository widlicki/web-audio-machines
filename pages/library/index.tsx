import { Grid } from '@mui/material'
import type { NextPage } from 'next'
import withRequest from '../../components/hoc/withRequest'
import AudioCategoryList from '../../components/machines/AudioCategoryList/AudioCategoryList'
import { useGetData } from '../../hooks/useGetData/useGetData'
import { AudioCategory } from '../../interfaces/library'
import { API_URL, AUDIO_CATEGORIES } from '../../networkRequests/constants'
import theme from '../../styles/Theme'

const AudioCategoriesWithRequest = withRequest(AudioCategoryList)

const LibraryPage: NextPage = () => {
  const {
    data: audioCategories,
    isLoading: isAudioCategoriesLoading,
    isError: isAudioCategoriesError,
  } = useGetData<AudioCategory[]>({
    requestUrl: `${API_URL}${AUDIO_CATEGORIES}`,
  })

  return (
    <Grid container spacing={2} sx={{ height: '100vh' }}>
      <Grid
        item
        xs={4}
        sx={{
          border: 1,
          borderRadius: 4,
          borderColor: theme.border.color.light,
        }}
      >
        <AudioCategoriesWithRequest
          categories={audioCategories}
          isLoading={isAudioCategoriesLoading}
          isError={isAudioCategoriesError}
        />
      </Grid>
      <Grid item xs={4}>
        test
      </Grid>
    </Grid>
  )
}

export default LibraryPage
