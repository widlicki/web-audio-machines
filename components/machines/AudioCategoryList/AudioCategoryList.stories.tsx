import { ComponentMeta, ComponentStory } from '@storybook/react'
import { mockAudioCategories } from '../../../mocks/library'
import AudioCategoryList from './AudioCategoryList'

export default {
  title: 'Machines/AudioCategoryList',
  component: AudioCategoryList,
} as ComponentMeta<typeof AudioCategoryList>

const Template: ComponentStory<typeof AudioCategoryList> = (args) => (
  <AudioCategoryList {...args} />
)

export const WithData = Template.bind({})
WithData.args = {
  categories: mockAudioCategories,
}

export const WithoutData = Template.bind({})
WithoutData.args = {
  categories: [],
}
