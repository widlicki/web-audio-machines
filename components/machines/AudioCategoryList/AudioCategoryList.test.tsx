import { render } from '@testing-library/react'
import { mockAudioCategories } from '../../../mocks/library'
import AudioCategoryList, { AudioCategoryListProps } from './AudioCategoryList'
import { WithData, WithoutData } from './AudioCategoryList.stories'

let component: any

const defaultArgs: AudioCategoryListProps = {
  categories: [],
} as AudioCategoryListProps

describe('AudioCategoryList Stories', () => {
  new Map([
    ['correct list when data exists', WithData],
    ['correct text when data does not exist', WithoutData],
  ]).forEach(({ args }, key) => {
    it(`renders ${key}`, () => {
      component = <AudioCategoryList {...defaultArgs} {...args} />
      const { container } = render(component)
      expect(container).toMatchSnapshot()
    })
  })
})

describe('AudioCategoryList', () => {
  beforeEach(() => {
    component = <AudioCategoryList categories={mockAudioCategories} />
  })

  it('Should render correct categories', () => {
    const { getByText } = render(component)

    mockAudioCategories.forEach((cat) => {
      expect(getByText(cat.name)).toBeTruthy()
    })
  })
})
