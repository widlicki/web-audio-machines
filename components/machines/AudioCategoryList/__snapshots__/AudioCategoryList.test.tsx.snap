// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioCategoryList Stories renders correct list when data exists 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ykqd4v-MuiListItem-root"
      >
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FolderIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Kick
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ykqd4v-MuiListItem-root"
      >
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FolderIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Bass
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ykqd4v-MuiListItem-root"
      >
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FolderSpecialIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 6h-8l-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-2.06 11L15 15.28 12.06 17l.78-3.33-2.59-2.24 3.41-.29L15 8l1.34 3.14 3.41.29-2.59 2.24.78 3.33z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Synth
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
      </li>
    </ul>
  </div>
</div>
`;

exports[`AudioCategoryList Stories renders correct text when data does not exist 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-wisbe8-MuiTypography-root"
      >
        - No Data -
      </p>
    </ul>
  </div>
</div>
`;
